src/api/users.ts

```diff
@@ -32,3 +32,47 @@ router.delete("/users/history/:courseId", async (req, res) => {
   await sql`DELETE FROM courses_completed WHERE user_id = ${user.id} AND course_id = ${courseId}`;
   res.send();
 });
+
+router.get("/users/progress/:planId", async (req, res) => {
+  const user = await authUser(req);
+  const planId = req.params.planId;
+
+  const result = await sql.begin(async (sql) => {
+    const planDetails = await sql`
+      SELECT pr.total_units, pr.required_courses
+      FROM plans p
+      JOIN programs pr ON pr.id = p.program_id
+      WHERE p.id = ${planId} AND p.user_id = ${user.id}`;
+
+    const { totalUnits, requiredCourses } = planDetails[0];
+
+    const completedCourses = await sql`
```

Comment:
You're getting unsafe `any` errors because you need to type the return value of the SQL statement.

See the courses endpoint for an example: https://github.com/DegreePrerequisiteTracker/Degree-Prerequisite-Tracker/blob/main/src/api/courses.ts#L23